src/realm/exec/realm2json.cpp

```diff
@@ -2,35 +2,71 @@
 #include <iostream>
 #include <realm/history.hpp>
 
+const char* legend = "Simple tool to output the JSON representation of a Realm:\n"
+                     "  realm2json [--link-depth=N] [--xjson] <.realm file>\n"
+                     "\n"
+                     "Options:\n"
+                     " --link-depth: How deep to traverse linking objects (use -1 for infinite). See test_json.cpp "
+                     "for more details. Defaults to 0.\n"
+                     " --xjson: Output should be formatted as MongoDB XJSON \n"
+                     "\n";
+
+void abort2(bool b, const char* fmt, ...)
```

Comment:
We should give this a more descriptive name like `abort_if`. Also, let's give arguments descriptive names - like `cond` instead of `b` to indicate that it's the condition on which to abort. Finally, this is a C function rather than an idiomatic C++ function. How about:

```
template <typename FormatStr>
void abort_if(bool cond, FormatStr fmt) {
    if (!cond) {
        return;
    }

    fputs(fmt, stderr);
    std::exit(1);
}

template <typename FormatStr, typename ... Args>
void abort_if(bool cond, FormatStr fmt, Args... args) {
    if (!cond) {
        return;
    }

    fprintf(stderr, fmt, args...);
    std::exit(1);
}
```

Reply:
Sounds good, will change. Again, I did this because I looked at a few of the tools in the repo that all use this pattern